subsys/bluetooth/audio/stream.c

```diff
@@ -264,10 +264,12 @@ void bt_audio_stream_reset(struct bt_audio_stream *stream)
 
 	BT_DBG("stream %p", stream);
 
-	if (stream == NULL || stream->conn == NULL) {
+	if (stream == NULL) {
 		return;
 	}
 
+	bt_audio_stream_detach(stream);
+
```

Comment:
I don't think it matters, but wouldn't it be more correct to terminate the CIG prior to detaching the ACL? I think there can be a race between ACL disconnect and the CIG cleanup. If the ACL is removed you will also get an ISO disconnect - hence remove stream->iso (I think).
By the way - why do we terminate the CIG when a stream is reset? a CIG can have multiple streams right? Hence the others might still need to run.

Reply:
> I don't think it matters, but wouldn't it be more correct to terminate the CIG prior to detaching the ACL?

CIGs are not related to ACLs, so there shouldn't be a problem there. A CIG can however not be terminated if there's any active CIS, so there might be a possible issue there. 

> By the way - why do we terminate the CIG when a stream is reset? a CIG can have multiple streams right? Hence the others might still need to run.

Poor designed and implemented termination procedure is the answer here I'm afraid :(

I've wanted to fix this for a while, but I see that I've forgotten to add an issue for it. The CIG should not be termianted based on stream/endpoint states, as it is an entirely separate entity that has nothing to do with any connections.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/44307